### ISSUE TITLE
allow to pass multiple documents to explain_prediction

### ISSUE THREAD

**Issue by kmike** [TeamHG-Memex/eli5#225]
Currently we allow to pass a single document to explain_prediction. I think we should allow to pass multiple documents as well.

There are two ways to handle multiple documents; both of them look useful. 

1) just show explanations for all instances - something similar to https://github.com/TeamHG-Memex/eli5/issues/205.

2) aggregate resulting explanations in some way. For example, https://github.com/andosa/treeinterpreter has this feature - you can check which features are important for predicting labels for a part of dataset. I'm not sure we should show all individual documents in this case; probably there should be an option to turn it off (or maybe there is already an option - can we turn it off using `show` argument?).

It may also help with https://github.com/TeamHG-Memex/eli5/issues/213#issuecomment-308677391.

**Comment by lopuhin** [comment]
I wrote a PoC for (2) outside of eli5, targeting only XGBoost. It worked much faster than calling ``explain_prediction`` multiple times and then aggregating the results. The only caveat is that I needed to do special handling of missing values: xgboost might use some positive feature and if it's missing, will make the score lower. The weight for this missing positive feature would be negative in this case - so I excluded them from aggregation and the feature weights started to make sense. But it might be that this handling is not always desired. This approach also requires custom code in each classifier we want to support, although it's not a huge change and one document can be seen as a special case of multiple documents here.
